webapp/src/test/java/io/github/microcks/util/ai/AICopilotHelperTest.java

```diff
@@ -422,4 +427,441 @@ public void testParseRequestResponseOutputSplitWithJson() {
+
+    // Create a ObjectMapper
+    ObjectMapper mapper = new ObjectMapper();
+    // Create a test JsonNode with references and tags
+    JsonNode specNode = mapper.readTree(jsonSpecNode);
+    JsonNode expectedSpecNode = mapper.readTree(jsonExpectedSpecNodeResult);
+
+    // Call the resolveReferenceAndRemoveTagsInNode method
+    AICopilotHelper.resolveReferenceAndRemoveTagsInNode(specNode, specNode);
+
+    // Verify that the references and tags are removed correctly
+    assertEquals(expectedSpecNode, specNode);
```

Comment:
Is it robust to compare raw json like this? What about using something JSONAssert? I think we already got it as a test dependency.

Reply:
I understand that, but since the resolveReferenceAndRemoveTagsInNode method (renamed resolveReferenceAndRemoveTokensInNode) mutates the specNode object and does not return an object to compare, it is necessary to create a JsonNode in the test method. Now, I modified the validation to be done with String objects instead of JsonNode objects.